### DISCUSSION TITLE
Filtering options

### DISCUSSION THREAD

**Discussion by pieeye75** [hnolCol/instantclue#58] [Q&A]
Hi all,

cool program, thanks for sharing!! I have a question on filtering. I have a stack of intensity values and these are in cell line categories. The problem is I have a lot of cell lines, too many to plot onto one bar graph. What is the best way to filter this data? I suspect my data is in the wrong format, but I'm still unsure how I would best go about doing this. Say I have 200 cell lines, each with 6 intensity values for the protein on interest (ie. 6 MS runs). I want to average the intensity for each run (6) and plot that for each cell line. No problem, I can do that, but the X-axis is unreadable with 200 cell lines listed. What is the best way to get the X-axis down to say 20 cell lines without manually removing all the other data?

thanks

**Comment by hnolCol** [comment]
Hi, 

thanks for starting the discussion. 
I think that you have two options. 
I assume the data look after transforming something like this:
![image](https://user-images.githubusercontent.com/24191910/188385694-2c4878df-d094-4706-b5ab-4fef980f1c90.png)

# Option 1 - using QuickSelect - Masking

First drag the categorical column to quick select:
![image](https://user-images.githubusercontent.com/24191910/188386497-afa338c3-7ccd-471b-a371-250b7b93b368.png)

and select "Unique values .." If you have no duplicated cell line names, selecting _raw values_ and _unique values .._ will give the same result. Then select the _masking mode_ as shown below.
![image](https://user-images.githubusercontent.com/24191910/188387476-ad77d293-d5c7-4e7f-8c0b-21697d74cfae.png)

Now you can select the cell lines you want to show. Non-selected values are masked and not plotted. You will have to click on the selected plot type again to make filtering visible. It does not automatically update the plot. 

# Option 2 - using factors 

Another option that could enable you to plot all cell line data but highlight specific ones.
You can also transfer cell line categories to numbers by using the _Factorize column(s)_ from the context menu.
![image](https://user-images.githubusercontent.com/24191910/188388678-0ff0f5b2-23da-41d8-a23a-159b1effff4c.png)
Then transform the new column _fac:(Cell Line)_ column to Numeric Floats and plot a scatter plot. You can then sort along the mean protein intensity and label some data-points (e.g. cell lines) that are of interest using the _Color_ _Size_ and _Labels_ options. Of note, if you put the Quick select mode back to _annotation_ you can label specific cell lines that will appear in every plot that you generate. 

If option 2 is of interest but the description is somewhat vague please let me know and I will provide more details. 


Cheers

Hendrik

**Comment by pieeye75** [comment]
Hi Hendrik,

thanks for the detailed instructions, Option 1 works well! I had a play with Option 2, but after factorising the cell lines, I was not able to transfer (??) the fac:(Cell Line) to the Numeric Floats. Did I miss something? I'm only starting to really get into Instant Clue, but it's clearly very powerful and saves me learning python! I need to play more to work out what it can do. I'm going through the tutorials.

One more question/suggestion. Clearly the calculations for the plotting are performed. Is there a way to output the results as a text file (for example)? ie. it would be great to have the plot data as a table.

thanks again!

Peter

------ Original Message ------

**Reply by hnolCol** [reply]
Hi Peter, 

happy that option 1 worked well for you. Please try the new release just updated today if the transformation to Numeric Floats works for the factorized column. 

To your second question, you can get for the Boxplots the so called "Summary data" by right-clickin on the figure and then select "Show summary data" which contains all the information required for the box. 

For violin and boxenplots, this is not available. Also not for barplots at the moment (in this case you would like to get the mean + error for example?). 

Best wishes

Hendrik
